v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstobigquery/BigtableChangeStreamsToBigQuery.java

```diff
@@ -0,0 +1,457 @@
+
+    PCollectionTuple tableRowTuple =
+        failsafeModJson.apply("Mod JSON To TableRow", failsafeModJsonToTableRow);
+
+    WriteResult writeResult =
+        tableRowTuple
+            .get(failsafeModJsonToTableRow.transformOut)
+            .apply(
+                "Write To BigQuery",
+                BigQueryIO.<TableRow>write()
+                    .to(bigQuery.getDynamicDestinations())
+                    .withFormatFunction(
+                        BigtableChangeStreamsToBigQuery::removeIntermediateMetadataFields)
+                    .withFormatRecordOnFailureFunction(element -> element)
+                    .withCreateDisposition(CreateDisposition.CREATE_IF_NEEDED)
+                    .withWriteDisposition(WriteDisposition.WRITE_APPEND)
+                    .withExtendedErrorInfo()
+                    .withMethod(Write.Method.STREAMING_INSERTS)
+                    .withFailedInsertRetryPolicy(InsertRetryPolicy.retryTransientErrors()));
+
+    PCollection<String> transformDlqJson =
+        tableRowTuple
+            .get(failsafeModJsonToTableRow.transformDeadLetterOut)
+            .apply(
+                "Failed Mod JSON During Table Row Transformation",
+                MapElements.via(new StringDeadLetterQueueSanitizer()));
+
+    PCollection<String> bqWriteDlqJson =
+        writeResult
+            .getFailedInsertsWithErr()
+            .apply(
+                "Failed Mod JSON During BigQuery Writes",
+                MapElements.via(new BigQueryDeadLetterQueueSanitizer()));
+
+    PCollectionList.of(transformDlqJson)
+        .and(bqWriteDlqJson)
+        .apply("Merge Failed Mod JSON From Transform And BigQuery", Flatten.pCollections())
+        .apply(
+            "Write Failed Mod JSON To DLQ",
+            DLQWriteTransform.WriteDLQ.newBuilder()
+                .withDlqDirectory(dlqDirectory)
+                .withTmpDirectory(tempDlqDirectory)
+                .setIncludePaneInfo(true)
+                .build());
+
+    PCollection<FailsafeElement<String, String>> nonRetryableDlqModJsonFailsafe =
+        dlqModJson.get(DeadLetterQueueManager.PERMANENT_ERRORS).setCoder(FAILSAFE_ELEMENT_CODER);
+
+    nonRetryableDlqModJsonFailsafe
+        .apply(
+            "Write Mod JSON With Non-retryable Error To DLQ",
+            MapElements.via(new StringDeadLetterQueueSanitizer()))
+        .setCoder(StringUtf8Coder.of())
+        .apply(
+            DLQWriteTransform.WriteDLQ.newBuilder()
+                .withDlqDirectory(dlqManager.getSevereDlqDirectoryWithDateTime())
+                .withTmpDirectory(dlqManager.getSevereDlqDirectory() + "tmp/")
+                .setIncludePaneInfo(true)
+                .build());
+
+    return pipeline.run();
+  }
+
+  private static String getChangeStreamName(BigtableChangeStreamToBigQueryOptions options) {
+    if (StringUtils.isBlank(options.getBigtableChangeStreamName())) {
+      String generatedName = UUID.randomUUID().toString();
```

Comment:
Can we let the connector default this instead? We recently updated it to use a shorter UID to save metadata table space

Reply:
done